source/particle/output/hdf5.cc

```diff
@@ -0,0 +1,256 @@
+        (void)current_time;
+
+        std::string             output_file_prefix, output_path_prefix, full_filename;
+
+        output_file_prefix = "particle-" + Utilities::int_to_string (file_index, 5);
+        output_path_prefix = this->get_output_directory() + output_file_prefix;
+#ifdef DEAL_II_HAVE_HDF5
+        // TODO: error checking for H5 calls
+        hid_t   h5_file_id;
+        hid_t   plist_id, xfer_plist_id, position_dataset_id, velocity_dataset_id, pid_dataset_id;
+        hid_t   file_dataspace_id, pos_file_dataspace_id, vel_file_dataspace_id, pid_file_dataspace_id;
+        hid_t   dim_dataspace_id, one_dim_ds_id;
+        hid_t   dim_mem_ds_id, one_dim_mem_ds_id;
+        hid_t   pattr_id;
+        hsize_t dims[2], offset[2], count[2];
+        unsigned int  mpi_offset, mpi_count, local_particle_count, global_particle_count, d, i;
+
+        double  *pos_data, *vel_data, *id_data;
+
+        std::string h5_filename = output_path_prefix+".h5";
+
+        // Create parallel file access
+        plist_id = H5Pcreate(H5P_FILE_ACCESS);
+#ifdef H5_HAVE_PARALLEL
+        H5Pset_fapl_mpio(plist_id, this->get_mpi_communicator(), MPI_INFO_NULL);
+#endif
+
+        // Create the file
+        h5_file_id = H5Fcreate(h5_filename.c_str(), H5F_ACC_TRUNC, H5P_DEFAULT, plist_id);
+
+        // Create the file dataspace descriptions
+        local_particle_count = particles.size();
+        // TODO: error checking on MPI call
+        MPI_Comm com = this->get_mpi_communicator();
+        MPI_Allreduce(&local_particle_count, &global_particle_count, 1, MPI_UNSIGNED, MPI_SUM, com);
+        dims[0] = global_particle_count;
+        dims[1] = 3;
+        one_dim_ds_id = H5Screate_simple(1, dims, NULL);
+        dim_dataspace_id = H5Screate_simple(2, dims, NULL);
+
+        // Create the datasets
+#if H5Dcreate_vers == 1
+        position_dataset_id = H5Dcreate(h5_file_id, "nodes", H5T_NATIVE_DOUBLE, dim_dataspace_id, H5P_DEFAULT);
+        velocity_dataset_id = H5Dcreate(h5_file_id, "velocity", H5T_NATIVE_DOUBLE, dim_dataspace_id, H5P_DEFAULT);
+        pid_dataset_id = H5Dcreate(h5_file_id, "id", H5T_NATIVE_DOUBLE, one_dim_ds_id, H5P_DEFAULT);
+#else
+        position_dataset_id = H5Dcreate(h5_file_id, "nodes", H5T_NATIVE_DOUBLE, dim_dataspace_id, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+        velocity_dataset_id = H5Dcreate(h5_file_id, "velocity", H5T_NATIVE_DOUBLE, dim_dataspace_id, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+        pid_dataset_id = H5Dcreate(h5_file_id, "id", H5T_NATIVE_DOUBLE, one_dim_ds_id, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+#endif
+
+        // Close the file dataspaces
+        H5Sclose(dim_dataspace_id);
+        H5Sclose(one_dim_ds_id);
+
+        // Just in case we forget what they are
+        count[0] = 1;
+        file_dataspace_id = H5Screate_simple (1, count, NULL);
+#if H5Acreate_vers == 1
+        pattr_id = H5Acreate(h5_file_id, "Ermahgerd! Pertecrs!", H5T_NATIVE_INT, file_dataspace_id, H5P_DEFAULT);
```

Comment:
This was the one class I never touched, because I do not know much about hdf5. After looking through it, I will change it substantially. I did not know about "Ermahgerd" before :-) .